dev-tools/reconfigurator-cli/src/main.rs

```diff
@@ -76,13 +81,66 @@ struct ReconfiguratorSim {
     /// External DNS zone name configured
     external_dns_zone_name: String,
 
+    /// RNG for collection IDs
+    collection_id_rng: TypedUuidRng<CollectionKind>,
+
     /// Policy overrides
     num_nexus: Option<u16>,
 
     log: slog::Logger,
 }
 
 impl ReconfiguratorSim {
+    fn new(log: slog::Logger) -> Self {
+        Self {
+            system: SystemDescription::new(),
+            collections: IndexMap::new(),
+            blueprints: IndexMap::new(),
+            internal_dns: BTreeMap::new(),
+            external_dns: BTreeMap::new(),
+            silo_names: vec!["default-silo".parse().unwrap()],
+            external_dns_zone_name: String::from("oxide.example"),
+            collection_id_rng: TypedUuidRng::from_entropy(),
+            num_nexus: None,
+            log,
+        }
+    }
+
+    /// Returns true if the REPL needs to be reset before loading an example.
+    ///
+    /// Loading example systems requires a (mostly) clean slate, which this
+    /// method checks for.
+    fn needs_reset_for_load_example(&self) -> bool {
```

Comment:
Take this one or leave it, but when it was named and documented as "is a reset needed" I found myself wondering "well when and why is a reset needed".

```suggestion
    /// Returns true if the user has made local changes to the simulated system
    ///
    /// This is used when the user asks to load an example system.  Doing that basically requires a clean slate.
    fn user_made_local_system_changes(&self) -> bool {
```

Reply:
Changed to `user_made_system_changes`. Though in my followup this just becomes `SimSystem::is_empty`.